muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceSystemRESTService.java

```diff
@@ -66,10 +75,56 @@
 
   @Inject
   private PluginSettingsController pluginSettingsController;
+
+  @Inject
+  private EnvironmentUserController environmentUserController;
+  
+  @Inject
+  private FileAnswerUtils fileAnswerUtils;
+  
+  @Inject
+  private WorkspaceMaterialFileFieldAnswerDAO workspaceMaterialFileFieldAnswerDAO;
+  
+  @GET
+  @Path("/movefileanswers")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response moveFileAnswers(@QueryParam("count") Integer count) {
+    if (count == null || count < 0) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    EnvironmentUser user = environmentUserController.findEnvironmentUserByUserEntity(sessionController.getLoggedUserEntity());
+    if (user == null || user.getRole() == null || user.getRole().getArchetype() != EnvironmentRoleArchetype.ADMINISTRATOR) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    int bytes = 0;
+    int totalBytes = 0;
+    int totalFiles = 0;
+    Long currentId = fileAnswerUtils.getLastEntityId();
+    List<Long> answerIds = workspaceMaterialFileFieldAnswerDAO.listIdsByLargerAndLimit(currentId, count);
+    for (Long answerId : answerIds) {
+      try {
+        WorkspaceMaterialFileFieldAnswer answer = workspaceMaterialFileFieldAnswerDAO.findById(answerId);
+        if (answer != null) {
+          bytes = fileAnswerUtils.relocateToFileSystem(answer);
+          if (bytes > 0) {
+            totalBytes += bytes;
+            totalFiles++;
+          }
+          currentId = answerId;
+        }
+      }
+      catch (IOException e) {
+        logger.log(Level.SEVERE, String.format("Failed to relocate WorkspaceMaterialFileFieldAnswer %d", currentId), e);
+        return Response.status(Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
+      }
+    }
+    fileAnswerUtils.setLastEntityId(currentId);
```

Comment:
Can there be problems if batch is too big and the operation times out (does it time out though - maybe database timeout?)? At least the last id is not updated.